src/components/login/login.tsx

```diff
@@ -54,22 +88,99 @@ class Login extends React.Component<Props, DefaultState> {
         });
     };
 
+    public handlePassphraseChange = (element) => {
+        this.setState({
+            passphrase: element.target.value,
+        });
+    };
+
+    public handleFilePathChange = (element) => {
+        this.setState({
+            filePath: element.target.value,
+        });
+    };
+
     public handleAgreementChange = (element, data) => {
         this.setState({
             agreement: data.checked ? 1 : 0,
         });
     };
 
-    public handleLogin = () => {
-        if (this.state.agreement) {
-            this.props.login(this.state.publicKey, this.state.privateKey);
-            return;
+    public handleLogin = async () => {
+        logger.info("Logging in user..");
+
+        try {
+            // verify keypair-relation
+            const keyPair = await this.cryptoKeyUtil.verifyKeyPair(
```

Comment:
I removed the variable assignment, because it isn't necessary. The function call breaks if someone supplies incorrect credentials and the catch clause handles this.